firmware/Core/Src/telecommands/telecommand_executor.c

```diff
@@ -240,4 +251,65 @@ uint8_t TCMD_agenda_delete_by_tssent(uint64_t tssent) {
         tssent_str
     );
     return 1;
+}
+
+/// @brief Fetches the active agendas.
+/// @return 0 on success, 1 if there are no active agendas.
+uint8_t TCMD_agenda_fetch(){
+    uint16_t active_agendas = 0;
+    uint16_t logged_agendas = 0;
+    char message_buffer[512];
+
+    // Count the number of active agendas
+    for (uint16_t slot_num = 0; slot_num < TCMD_AGENDA_SIZE; slot_num++) {
+        if (TCMD_agenda_is_valid[slot_num]) {
+            active_agendas++;
+        }
+    }
+
+    // if no active agendas, return 1
+    if(active_agendas == 0){
+        LOG_message( LOG_SYSTEM_TELECOMMAND, LOG_SEVERITY_NORMAL, LOG_SINK_ALL,
+        "TCMD_agenda_fetch: No entries in the agenda."
+    );
+    return 1;
+    }
+
+    // Output the number of active agendas
+    LOG_message(
+        LOG_SYSTEM_TELECOMMAND, LOG_SEVERITY_NORMAL, LOG_SINK_ALL,
+        "TCMD_agenda_fetch: Active agendas: %u",
+        active_agendas
+    );
+
+    // List all active agendas in JSONL format
+    for (uint16_t slot_num = 0; slot_num < TCMD_AGENDA_SIZE; slot_num++) {
+        if (TCMD_agenda_is_valid[slot_num]) {
+
+            // Convert uint64_t to a string
+            char tssent_str[32];
+            GEN_uint64_to_str(TCMD_agenda[slot_num].timestamp_sent, tssent_str);
+            char tsexec_str[32];
+            GEN_uint64_to_str(TCMD_agenda[slot_num].timestamp_to_execute, tsexec_str);
+
+            snprintf(
+                message_buffer, sizeof(message_buffer),
+                "{\"slot_num\":\"%u\",\"telecommand_channel\":\"%s\",\"timestamp_sent\":%s,\"timestamp_to_execute\":%s}\n",
+                slot_num,
+                getTelecommandChannelName(TCMD_agenda[slot_num].tcmd_channel),
+                tssent_str,
+                tsexec_str
+            );
+        }
+            LOG_message( LOG_SYSTEM_TELECOMMAND, LOG_SEVERITY_NORMAL, LOG_SINK_ALL, message_buffer);
+            logged_agendas++;
+
+            // Break the loop once all active agendas have been logged
```

Comment:
Please note that this is an "early-exit optimization" in a comment

Reply:
Sure thing. I added the early-exit optimization into the comment